priiloader/source/hacks.cpp

```diff
@@ -549,14 +563,62 @@ s8 LoadSystemHacks(bool load_nand)
 
 			mem_free(fbuf);
 
-			//Set all system_hacks for system menu > max_version || sysver < min_version to 0
+			
 			unsigned int sysver = GetSysMenuVersion();
+			
+			// Make sure that all master hacks are disabled in the state array. 
+			// That way, when all sub-hacks are disabled, the master hack will 
+			// be disabled as well. Then, in the next loop, if a sub hack is 
+			// selected, the master hack will be enabled again. 
+			for( u32 i=0; i < system_hacks.size(); ++i)
+			{
+				if (system_hacks[i].provide.length() > 0) {
+					// This is a master hack
+					states_hash[i] = 0; 

```

Comment:
The point of the master / slave system was that the master hack is hidden from the user, and each sub hack enables the master + specific changes. I always disable the master hacks here, otherwise there would be no way for the user to disable a hidden master hack. 

This way, the master hack is disabled every time, and only enabled when a matching sub hack is enabled.

Reply:
ok granted, a master alone is probably not done.
however, my comment stands that having the check to disable it is useless. the master should only be disabled if all it's subs are disabled. 
im sure there is a fast way to do that, probably using lambda

Reply:
Lambda, with all its setup stuff, the iterator and such, is probably slower than a simple for loop, so if you are looking for raw performance, a loop would probably be even better. And, in my opinion, more readable, too. 
It's your application so if you absolutely want it done with a lambda expression I will do that - but it's not going to be faster than a loop. It's just syntactic sugar plus more setup stuff for a loop. 


If I wanted to disable a master when no sub is selected, that would take even more nested loops. I would need to go through the list, finding each master hack; and then for each master hack, go through the list again and check if there is any sub hack that requires said master, and if not, disable it. Way slower than always disabling it and then enabling it when needed, in my opinion. 

Am I missing something here? Is there an easier / more performant way for that?